presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1885,89 +1885,213 @@ private void computeAndAssignOrderByScopeWithAggregation(OrderBy node, Scope sou
+
+                QualifiedName prefix = asQualifiedName(expression);
+                if (prefix != null) {
+                    // analyze prefix as an 'asterisked identifier chain'
+                    Optional<AsteriskedIdentifierChainBasis> identifierChainBasis = scope.resolveAsteriskedIdentifierChainBasis(prefix);
+                    if (!identifierChainBasis.isPresent()) {
+                        throw semanticException(TABLE_NOT_FOUND, allColumns, "Unable to resolve reference " + prefix);
+                    }
+                    if (identifierChainBasis.get() == AMBIGUOUS) {
+                        throw semanticException(AMBIGUOUS_IDENTIFIER_CHAIN, allColumns, "Reference '%s' is ambiguous", prefix);
+                    }
+                    if (identifierChainBasis.get() == TABLE) {
+                        // TODO enable all fields reference from outer scope relation
+                        Optional<RelationType> relationType = scope.resolveRelationTypeWithAlias(prefix, true);
+                        // TODO when outer scope relations are supported, do checkState(relationType.isPresent(), "missing relation type"); instead
+                        if (!relationType.isPresent()) {
+                            throw semanticException(NOT_SUPPORTED, allColumns, "SELECT * from outer scope relation not yet supported");
                         }
-                        if (target instanceof Identifier) {
-                            return QualifiedName.of(ImmutableList.of((Identifier) target));
+                        List<Field> fields = relationType.get().resolveFieldsWithRelationAlias(prefix).stream()
+                                .filter(field -> !field.isHidden())
+                                .collect(toImmutableList());
+                        if (fields.isEmpty()) {
+                            throw semanticException(COLUMN_NOT_FOUND, allColumns, "SELECT * not allowed from relation that has no columns");
                         }
-                        throw semanticException(NOT_SUPPORTED, allColumns, "Only identifier chains supported with .*");
-                    });
-
-            if (!allColumns.getAliases().isEmpty()) {
-                throw semanticException(NOT_SUPPORTED, allColumns, "Column aliases not supported");
+                        analyzeAllColumnsFromTable(fields, allColumns, node, scope, outputExpressionBuilder, selectExpressionBuilder, relationType.get());
+                    }
+                    else { // identifierChainBasis.get() == FIELD
+                        analyzeAllFieldsFromRowTypeExpression(expression, allColumns, node, scope, outputExpressionBuilder, selectExpressionBuilder);
+                    }
+                }
+                else {
+                    // analyze all cases of expression.* other than QualifiedName.*
+                    analyzeAllFieldsFromRowTypeExpression(expression, allColumns, node, scope, outputExpressionBuilder, selectExpressionBuilder);
```

Comment:
Instead of repeating this, I'd change the structure of the code slightly:

```java
if (allColumns.getTarget().isPresent()) {
    ...

    if (prefix != null) {
        ...

        if (identifierChainBasis.get() == TABLE) {
            ...
            analyzeAllColumnsFromTable(fields, allColumns, node, scope, outputExpressionBuilder, selectExpressionBuilder, relationType.get());
            return;
        }
    }
    analyzeAllFieldsFromRowTypeExpression(expression, allColumns, node, scope, outputExpressionBuilder, selectExpressionBuilder);
}
```